satpy/dataset.py

```diff
@@ -98,10 +98,21 @@ def combine_metadata(*metadata_objects, **kwargs):
     shared_info = {}
     for k in shared_keys:
         values = [nfo[k] for nfo in info_dicts]
-        any_arrays = any([isinstance(val, np.ndarray) for val in values])
+        any_arrays = any([hasattr(val, "__array__") for val in values])
+        # in the real world, the `ancillary_variables` attribute may be
+        # List[xarray.DataArray], this means our values are now
+        # List[List[xarray.DataArray]].
+        any_list_of_arrays = any(
+                [isinstance(val, Collection) and len(val) > 0 and
+                 all([hasattr(subval, "__array__")
+                     for subval in val])
+                 for val in values])
         if any_arrays:
             if all(np.all(val == values[0]) for val in values[1:]):
                 shared_info[k] = values[0]
+        elif any_list_of_arrays:
+            if all(np.array_equal(val, values[0]) for val in values[1]):
```

Comment:
This works, but it computes the arrays and it's therefore (very) slow.  I am interested in alternative ideas, in particular from @djhoese and @mraspaud, who I think the `combine_metadata` function so may be better positioned to consider its intention.

Reply:
Now comparing by object identity.